.github/workflows/api-server-test-Postgres.yml

```diff
@@ -0,0 +1,111 @@
+name: API Server Tests - Postgres
```

Comment:
I would like us to explore whether we can leverage the  existing `api-server-tests.yml`, this would allow us to continue to focus api tests in one workflow. Can we also use the `./.github/actions/test-and-report` action like the other jobs in `api-server-tests.yml`, this way we can continue to have good test reports generated for the postgresql case. 

If we need to ignore upgrade tests, we can update the action to take in optional `--label-filter` values 


cc @nsingla